src/components/Sponsors/Sponsors.tsx

```diff
@@ -1,9 +1,33 @@
 import React, { useState, useEffect } from 'react';
 import { useColorMode } from '@docusaurus/theme-common';
 import { sanitize } from 'dompurify';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+
+const cycleInterval = 10000; // 10 seconds
 
 export default function Sponsors(): React.JSX.Element {
   const [data, setData] = useState(null);
+
+  const [counter, setCounter] = useState(0);
+  const [seconds, setSeconds] = useState(0);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setCounter((prevCounter) => prevCounter + 1);
+    }, cycleInterval);
+
+    return () => clearInterval(interval);
+  }, []);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setSeconds((prevCounter) => prevCounter + 1);
+    }, cycleInterval / 5);
+
+    return () => clearInterval(interval);
+  }, []);
```

Comment:
```suggestion
  const timerCleanUp = (func: React.Dispatch<React.SetStateAction<number>>, interval: number): (() => void) => {
    const temp = setInterval(() => {
      func((prevCounter) => prevCounter + 1);
    }, interval);

    return () => clearInterval(temp);
  };

  useEffect(() => {
    return timerCleanUp(setCounter, cycleIntervalMs);
  }, []);

  useEffect(() => {
    return timerCleanUp(setSeconds, cycleIntervalMs / 5);
  }, []);

```
Just some refactoring to prevent code duplication